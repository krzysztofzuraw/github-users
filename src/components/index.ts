export { MainCard } from "@src/components/main-card";
export { List } from "@src/components/list";
export { LocaleCard } from "@src/components/locale-card";
