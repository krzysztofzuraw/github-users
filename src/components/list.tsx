import { Avatar, Button, List as AntList } from "antd";
import * as React from "react";
import { FormattedMessage } from "react-intl";
import { style as tss } from "typestyle/lib";

import { IUser } from "@src/models";

interface IProps {
  items: IUser[];
  loading: boolean;
  onReload: () => void;
}

const styles = {
  actions: tss({
    alignItems: "center",
    display: "flex",
    justifyContent: "center",
    marginTop: "10px",
  }),
};

export const List: React.SFC<IProps> = ({ items, loading, onReload }) => (
  <AntList
    itemLayout="horizontal"
    dataSource={items}
    loadMore={
      <div className={styles.actions}>
        <Button onClick={onReload} loading={loading}>
          <FormattedMessage id="list.button.reload" defaultMessage="Refresh" />
        </Button>
      </div>
    }
    loading={loading}
    renderItem={(item: IUser) => (
      <AntList.Item
        actions={[
          <a href={item.htmlUrl}>
            <FormattedMessage id="list.link.profile" defaultMessage="Profile" />
          </a>,
        ]}
      >
        <AntList.Item.Meta
          avatar={<Avatar src={item.avatarUrl} />}
          title={item.login}
          description={
            item.bio || (
              <FormattedMessage id="list.description.default" defaultMessage="No description" />
            )
          }
        />
      </AntList.Item>
    )}
  />
);
