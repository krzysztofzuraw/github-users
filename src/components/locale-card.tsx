import { Button, Card } from "antd";
import * as React from "react";
import { FormattedMessage } from "react-intl";
import { style } from "typestyle/lib";

interface IProps {
  onClick: (lang: string) => void;
  selectedLang: string;
}

interface IState {
  currentLang: string;
}

const styles = {
  card: style({
    marginTop: "20px",
  }),
  wrapper: style({
    display: "flex",
    justifyContent: "space-between",
  }),
};

export class LocaleCard extends React.Component<IProps, IState> {
  state = { currentLang: this.props.selectedLang };

  handleClick = (lang: string) => {
    this.setState({ currentLang: lang });
    this.props.onClick(lang);
  };

  getButtonType = (lang: string) => {
    const { currentLang } = this.state;
    return currentLang === lang ? "primary" : "default";
  };

  render() {
    const { handleClick, getButtonType } = this;
    return (
      <Card
        title={<FormattedMessage id="locale.card.title" defaultMessage="Change locale" />}
        className={styles.card}
      >
        <div className={styles.wrapper}>
          <Button shape="circle" type={getButtonType("en")} onClick={() => handleClick("en")}>
            EN
          </Button>
          <Button shape="circle" type={getButtonType("pl")} onClick={() => handleClick("pl")}>
            PL
          </Button>
        </div>
      </Card>
    );
  }
}
