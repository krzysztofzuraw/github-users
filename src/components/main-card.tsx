import { Avatar, Button, Card, List } from "antd";
import * as React from "react";
import { FormattedMessage } from "react-intl";

export const MainCard: React.SFC = () => {
  return (
    <Card title={<FormattedMessage id="main.card.header" defaultMessage="Who to follow" />}>
      <List
        itemLayout="horizontal"
        dataSource={["1", "2", "3"]}
        loadMore={
          <div>
            <Button>Refresh</Button>
          </div>
        }
        renderItem={(item: any) => (
          <List.Item actions={[<a>follow</a>]}>
            <List.Item.Meta
              avatar={
                <Avatar src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png" />
              }
              title="User Name"
              description="User description goes here"
            />
          </List.Item>
        )}
      />
    </Card>
  );
};
