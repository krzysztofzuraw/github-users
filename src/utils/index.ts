export { componentWithIntl, createMockStoreWithEpic } from "@src/utils/tests";
