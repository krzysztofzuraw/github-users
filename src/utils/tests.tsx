import * as React from "react";
import { IntlProvider } from "react-intl";

import * as enMessages from "@locales/en.json";

const componentWithIntl = (
  children: JSX.Element,
  props = { locale: "en", messages: { ...enMessages } }
) => <IntlProvider {...props}>{children}</IntlProvider>;

export default componentWithIntl;
