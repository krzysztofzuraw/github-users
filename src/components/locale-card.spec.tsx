import each from "jest-each";
import * as React from "react";
import { cleanup, render } from "react-testing-library";

import { LocaleCard } from "@src/components/locale-card";
import { componentWithIntl } from "@src/utils";

afterEach(cleanup);

describe("Locale card", () => {
  each([["pl"], ["en"]]).test("should render card when %s language is selected", (lang: string) => {
    const component = render(
      componentWithIntl(<LocaleCard selectedLang={lang} onClick={jest.fn()} />)
    );
    expect(component).toMatchSnapshot();
  });
});
