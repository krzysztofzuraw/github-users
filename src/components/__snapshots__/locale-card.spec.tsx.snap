// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locale card component should render card when en language is selected 1`] = `
Object {
  "baseElement": <body>
    <div>
      <div
        class="ant-card fzzj7d9 ant-card-bordered"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              <span>
                Change locale
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="f1ubk0hm"
          >
            <button
              class="ant-btn ant-btn-primary ant-btn-circle"
              type="button"
            >
              <span>
                EN
              </span>
            </button>
            <button
              class="ant-btn ant-btn-default ant-btn-circle"
              type="button"
            >
              <span>
                PL
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-card fzzj7d9 ant-card-bordered"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            <span>
              Change locale
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="f1ubk0hm"
        >
          <button
            class="ant-btn ant-btn-primary ant-btn-circle"
            type="button"
          >
            <span>
              EN
            </span>
          </button>
          <button
            class="ant-btn ant-btn-default ant-btn-circle"
            type="button"
          >
            <span>
              PL
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Locale card component should render card when pl language is selected 1`] = `
Object {
  "baseElement": <body>
    <div>
      <div
        class="ant-card fzzj7d9 ant-card-bordered"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              <span>
                Change locale
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="f1ubk0hm"
          >
            <button
              class="ant-btn ant-btn-default ant-btn-circle"
              type="button"
            >
              <span>
                EN
              </span>
            </button>
            <button
              class="ant-btn ant-btn-primary ant-btn-circle"
              type="button"
            >
              <span>
                PL
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-card fzzj7d9 ant-card-bordered"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            <span>
              Change locale
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <div
          class="f1ubk0hm"
        >
          <button
            class="ant-btn ant-btn-default ant-btn-circle"
            type="button"
          >
            <span>
              EN
            </span>
          </button>
          <button
            class="ant-btn ant-btn-primary ant-btn-circle"
            type="button"
          >
            <span>
              PL
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
